rust-escrow/conditional-escrow/src/lib.rs

```diff
@@ -171,19 +172,32 @@ impl ConditionalEscrow {
             recipient_contract_id.clone(), // contract ID
             total_funds,                   // funds
             GAS_FOR_DELEGATE,              // gas
-        );
-
-        self.total_funds = 0;
-
-        log!(
-            "Delegating {} NEAR tokens to {}. — Total funds held after call: {}",
-            total_funds,
-            recipient_contract_id,
-            self.get_total_funds()
-        );
+        )
+        .then(ext_self::on_create_dao_callback(
+            env::current_account_id(),
+            0,
+            GAS_FOR_DELEGATE_CALLBACK,
+        ))
         // @TODO emit delegate_funds event
     }
 
+    #[private]
+    pub fn on_create_dao_callback(&mut self) -> bool{
+        assert_eq!(env::promise_results_count(), 1, "ERR_CALLBACK_METHOD");
+
+        match env::promise_result(0) {
+            PromiseResult::Successful(result) => {
+                let res = String::from_utf8_lossy(&result);
+                if res == "true" {
+                    self.total_funds = 0;
+                    return true;
+                }
+                false
+            }
+            _ => false,
+        }
```

Comment:
Would this implementation work the same as this one? I think it is cleaner:
https://github.com/aufacicenta/near.holdings/blob/master/rust-escrow/src/lib.rs#L183-L199

```suggestion
        pub fn on_create_dao_callback(&mut self) -> bool {
        assert_eq!(env::promise_results_count(), 1, "ERR_CALLBACK_METHOD");

        if near_sdk::is_promise_success() {
            self.total_funds = 0;
            true
        } else {
            false
        }
    }
```